api/texts_api.py

```diff
@@ -0,0 +1,170 @@
+import django
+django.setup()
+from sefaria.utils.hebrew import hebrew_term
+from .api_errors import *
+from .helper import split_query_param_and_add_defaults
+from typing import List
+
+
+class VersionsParams():
+    """
+    an object for managing the versions params for TextsHandler
+    params can come from an API request or internal (sever side rendering)
+    lang is our code language. special values are:
+        source - for versions in the source language of the text
+        base - as source but with falling to the 'nearest' to source, or what we have defined as such
+    vtitle is the exact versionTitle. special values are:
+        no vtitle - the version with the max priority attr of the specified language
+        all - all versions of the specified language
+    representing_string is the original string that came from an API call
+    """
+
+    def __init__(self, lang: str, vtitle: str, representing_string=''):
+        self.lang = lang
+        self.vtitle = vtitle
+        self.representing_string = representing_string
+        if not self.representing_string:
```

Comment:
I think making this an input is really confusing. If we only need this string for one error message later, it should be taken care of in that context and not here.